### ISSUE TITLE
Suggestion field: A first suggestion is highlighted

### ISSUE THREAD

**Issue by axgurt** [cuba-platform/cuba#2710]
### Environment

- Platform version: 6.10.15
- Client type: Web

### Description of the bug or enhancement

[Suggestion_field__A_first_suggestion_is_highlited.zip](https://github.com/cuba-platform/cuba/files/4213237/Suggestion_field__A_first_suggestion_is_highlited.zip)

Steps to reproduce:
1. Type something into suggestion picker field.
Actual result: the first suggestion is selected by default as soon as it appears on the screen
Expected result: no suggestion shoud be selected by default

A test for reproducing this case has been attached.
Patch can be applied for https://git.haulmont.com/platform/platform-ui-test project.

**Comment by glebfox** [comment]
The described behavior isn't a bug. Most drop-down components behave the same way (not only in CUBA), e.g. `LookupField`:

<img width="254" alt="Screenshot 2020-09-09 at 12 55 25" src="https://user-images.githubusercontent.com/6077149/92577488-bf351080-f29b-11ea-94df-851adca292d0.png">

Selecting the first option in the search result prevents a user from extra actions to select an item.

As I can see from your patch, you added `EnterActionHandler`. `EnterActionHandler` is fired when the user isn't in the process of searching the value and mustn't take precedence over this action as it does in the current implementation. 

We won't change the current component behavior in CUBA. Let me know if you have a special case when you need to prevent the first suggestion to be selected by default, so I'll investigate is it possible to achieve this in a target project.

**Comment by dkz** [comment]
@glebfox I understand that this behavior is not a bug, however the fact that it can not be customized is beyond upsetting.
Use cases with suggestion meant to be deliberately chosen by user aren't really rare: in this particular case this field is used as address search field with familiar addresses suggestions.

**Comment by glebfox** [comment]
So, the real case is that you need to allow user input, i.e. if the user types something, this text is converted to the field value without the need to select one of the suggestions? Something like implementing `EnterActionHandler` as follows:

```
suggestionField.setEnterActionHandler(searchString -> {
    suggestionField.setValue(searchString);
});
```

If so, this makes much more sense that your sample with opening a screen on Enter press. 

Correct me if my assumption is wrong.

**Comment by dkz** [comment]
@glebfox the actual use scenario goes like this:

I (the user) want to type a text address into the field that shows my favorite addresses as suggestions.
If I need one of my fav addresses I choose a suggestion by pressing `arrow down` and selecting it manually, otherwise I simply want to perform an address search by pressing Enter without suggestion field interfering in the process.

An old suggestion field, our project forked and extended almost a decade ago, was working as expected in this scenario. Legacy component was replaced with the fresh CUBA one to leverage technical debt caused by possible major-version migrations, however it's behavior deviates from the original one.

**Comment by anipchenko** [comment]
Hi Gleb @glebfox 
Yesterday our client got a new Sherlock release (v49) based on Cuba 6.10.17 and complained about the issue with address search:
Matthew 21:10
Daniil apologies as I know it is late in Samara but it seems like Wayne has an issue with the address search 
it seems for some postcodes when he searches then presses ENTER it is not showing address search pop up box 
it is simply moving down to D/O search box 
example is wc1n3ln postcode 

Matthew 15:37
Hi Daniil, hope you are well today. 
I have spoken to Wayne this morning and he has said the following about the postcode search issue from yesterday "Also the issue with the post code is not just on the web version, as it was the desktop version that I was using and Lisa was using the web version"

@glebfox @knstvk Could you backport the solution from the linked ticket to Cuba 6.10.17?

**Comment by glebfox** [comment]
Hi @anipchenko, 

The requested changes available since **6.10.19**. We can't backport something to the released version. 

Regards,
Gleb
